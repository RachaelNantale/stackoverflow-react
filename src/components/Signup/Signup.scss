#container1 {
  // background-color: #e2dada;
  // display: flex;
  // align-items: center;
}

.centered-form {
  width: 70%;
  // display: flex;
  align-items: center;
  padding: 30px;
  border: solid 1px #cccccc;

  form {
    width: 100%;
  }
}

// .centered-form .panel {
//   background: rgba(255, 255, 255, 0.8);
//   box-shadow: rgba(0, 0, 0, 0.3) 20px 20px 20px;
// }