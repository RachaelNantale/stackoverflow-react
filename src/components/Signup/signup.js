import React from 'react';
import PropTypes from 'prop-types';
import './Signup.scss';

const Signup = ({ onSubmit, onChange, title }) => (
  <div className=" signup-container">
    <form className="centered-form" onSubmit={onSubmit}>
      <h3 className="text-center ">{title}</h3>
      <div className="form-group">
        <input
          type="email"
          className="form-control"
          id="email"
          name="email"
          placeholder="Email"
          onChange={onChange}
        />
      </div>
      <div className="form-group">
        <input
          type="password"
          className="form-control"
          id="password"
          placeholder="Password"
          name="password"
          onChange={onChange}
        />
      </div>
      <button type="submit" className="btn btn-primary" onClick={onSubmit}>Submit</button>
    </form>
  </div>

);
Signup.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
};

export default Signup;
