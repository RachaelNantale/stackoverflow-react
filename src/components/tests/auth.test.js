import React from 'react';
import { shallow } from 'enzyme';
import Signup from '../Signup/signup';


it('renders the signup component', () => {
  const signUpComponent = shallow(<Signup onSubmit={() => true} onChange={() => true} title="title" />);
  expect(signUpComponent).toMatchSnapshot();
});
