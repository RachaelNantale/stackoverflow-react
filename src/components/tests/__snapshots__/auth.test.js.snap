// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the signup component 1`] = `
<div
  className=" signup-container"
>
  <form
    className="centered-form"
    onSubmit={[Function]}
  >
    <h3
      className="text-center "
    >
      title
    </h3>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
